src/App.tsx

```diff
@@ -1,223 +1,217 @@
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React, { useContext, useEffect, useRef } from 'react';
+import React, {
+  useContext,
+  useEffect,
+  useMemo,
+  useRef,
+  useState,
+} from 'react';
+import { deleteTodo, getTodos, postTodo } from './api/todos';
 import { AuthContext } from './components/Auth/AuthContext';
+import { Filter } from './components/Filter';
+import { TodoList } from './components/TodoList';
+import { ErrorMessage } from './components/ErrorMessage';
+
+import { FilterStatus } from './types/FilterStatus';
+import { Todo } from './types/Todo';
+
+const getTodoById = (todos: Todo[], todoId: number) => {
+  return todos.find(({ id }) => id === todoId) || null;
+};
+
+const filterTodos = (todos: Todo[], filterStatus: FilterStatus) => {
```

Comment:
useMemo

Reply:
I implemented useMemo to get visibleTodos